numpy/core/src/umath/ufunc_object.c

```diff
@@ -5572,6 +5603,8 @@ ufunc_get_identity(PyUFuncObject *ufunc)
         return PyInt_FromLong(1);
     case PyUFunc_Zero:
         return PyInt_FromLong(0);
+    case PyUFunc_MinusOne:
```

Comment:
I think the only thing that might matter (not even sure about that) is, whether we want to return -1 or -1L in the object corner cases of empty reductions on python 2. But either way is good in any case, Python makes no real difference between the two (and will silently convert easily).